applications/asset_tracker/src/main.c

```diff
@@ -502,27 +503,36 @@ static void modem_rsrp_handler(char rsrp_value)
 		return;
 	}
 
-	k_work_submit_to_queue(&application_work_q, &rsrp_work);
+	/* Only send the RSRP if transmission is not already scheduled.
+	 * Checking CONFIG_HOLD_TIME_RSRP gives the compiler a shortcut.
+	 */
+	if (CONFIG_HOLD_TIME_RSRP == 0 ||
+	    k_delayed_work_remaining_get(&rsrp_work) == 0) {
+		k_delayed_work_submit_to_queue(&application_work_q, &rsrp_work,
+					       K_NO_WAIT);
+	}
```

Comment:
Suggestion, not request for change:
I was wondering if we in the future could keep track of last time we submitted this work, and then call it with a delay of `prev_timestamp - CONFIG_HOLD_TIME_RSRP` to avoid popping out of idle every CONFIG_HOLD_TIME_RSRP interval.

Reply:
We only pop once after each transmission, the behaviour is not self-sustaining. The savings would only come when there is only a single update in an interval. Nevertheless, that idea is clever. It does save on that wakeup, and all the scheduling could be done in the callback. I think we should leave this as-is and get it into 1.2 first, though.